### PR TITLE
[fix](function)When sublength in substr() is negative, will casue mem…

### PR DESCRIPTION
**Pull request by xuwei0912**
…ory out of bounds, same as left()

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -282,7 +282,7 @@ struct SubstringUtil {
             int start_value = is_const ? start[0] : start[i];
             int len_value = is_const ? len[0] : len[i];
 
-            if (start_value > str_size || start_value < -str_size || str_size == 0) {
+            if (start_value > str_size || start_value < -str_size || str_size == 0 || len_value <= 0) {
```
Comment (yiguolei) [inline be/src/vec/functions/function_string.h:285]: add regression test for this case

**Comment by xuwei0912** [comment]
run buildall

**Comment by yiguolei** [comment]
@xuwei0912  Does master branch have this problem?

**Comment by yiguolei** [comment]
run buildall

**Pull request merged**